src/libs/TransactionUtils/index.ts

```diff
@@ -1550,6 +1551,13 @@ function shouldShowRTERViolationMessage(transactions?: Transaction[]) {
     return transactions?.length === 1 && hasPendingUI(transactions?.at(0), getTransactionViolations(transactions?.at(0)?.transactionID, allTransactionViolations));
 }
 
+/**
+ * Return transactions pending action.
+ */
+function getTransactionPendingAction(transaction: OnyxEntry<Transaction>): PendingAction {
+    return transaction?.pendingAction ?? (Object.keys(transaction?.pendingFields ?? {}).length > 0 ? CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE : null);
```

Comment:
There is a lot of logic in this one line which makes it hard to read. I had to spent like 2 minutes to understand what happens 😅 

I suggest first run an if for the `transaction?.pendingAction` and return it, then run the rest of logic